### PR TITLE
mass loading module

### PR DESCRIPTION
**Pull request by caillef**

### CODE REVIEW

File: lua/modules/massloading.lua [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+massLoading.load = function(_, list, config)
+    local defaultConfig = {
+		onLoad = nil,
+		onListLoaded = nil,
+		fullnameItemKey = "fullname",
+	}
+    config = require("config"):merge(defaultConfig, config, {
+        acceptTypes = { onLoad = { "function" }, onListLoaded = { "function" } }
+    })
+	if not config.onLoad then
```
Comment (aduermael) [inline lua/modules/massloading.lua:55]: Are you using stylua?
There are a few indentation errors.

File: lua/modules/massloading.lua [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+		error("you must define config.onLoad")
+		return
+	end
+
+    local nbObjectsLoaded = 0
+    local function loadedNextObject()
+        nbObjectsLoaded = nbObjectsLoaded + 1
+        if nbObjectsLoaded >= #list then
+            awaitingObjects = {}
+            loadingObjects = {}
```
Comment (aduermael) [inline lua/modules/massloading.lua:65]: I don't think that's right. 
When done loading this list, it resets `awaitingObjects ` and `loadingObjects` while they could be shared with other loading processes.

**Review by aduermael — CHANGES_REQUESTED** [review]

File: lua/modules/massloading.lua (hunk 1 above)
Comment (caillef) [reply lua/modules/massloading.lua:55]: It seems that it was not formatting by default, just updated it

File: lua/modules/massloading.lua (hunk 2 above)
Comment (caillef) [reply lua/modules/massloading.lua:65]: That's right, I removed these two lines

**Review by aduermael — APPROVED** [review]
LGTM!

**Pull request merged**